### PR TITLE
Added the velvet dependency package and velvetoptimiser/velvet tools

### PR DESCRIPTION
**Pull request by Slugger70**
Adds package_velvet_1_2_10 and velvetoptimiser tool. Velvet Optimiser tool also contains a very basic velvet wrapper used for teaching effects of kmer size on assembly graphs.

### CODE REVIEW

**Comment by bgruening** [comment]
`VelvetOptimiser.pl: command not found`? @Slugger70 where is this file located?

**Comment by Slugger70** [comment]
I need to change FindBin::Bin to FindBin::RealBin so it resolves the symlink properly.

I need to patch VelvetOptimiser so it will work with conda

File: packages/package_velvet_1_2_10/tool_dependencies.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<tool_dependency>
+    <package name="velvet" version="1.2.10">
+        <install version="1.0">
+            <actions>
+                <action type="shell_command">git clone git://github.com/dzerbino/velvet.git</action>
```
Comment (yhoogstrate) [inline packages/package_velvet_1_2_10/tool_dependencies.xml]: It was more or less decided to use github urls in the following way:
 
 `<action type="download_by_url" sha256sum="8e952e3a4bf2ffae586a64e5e97029c9fdd53b5e1991106ef832d73e499b163b">https://github.com/dzerbino/velvet/archive/9adf09f7ded7fedaf6b0e5e4edf9f46602e263d3.tar.gz</action>`

File: tools/velvetoptimiser/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<tool_dependency>
+    <package name="velvet" version="1.2.10">
+        <repository name="package_velvet_1_2_10" owner="simon-gladman" />
+    </package>
+    <package name="velvetoptimiser" version="2.2.5">
```
Comment (yhoogstrate) [inline tools/velvetoptimiser/tool_dependencies.xml]: Does this need to be a separate package?

File: tools/velvetoptimiser/velvetoptimiser.xml [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    <requirements>
+      <requirement type="package" version="1.2.10">velvet</requirement>
+      <requirement type="package" version="2.2.5">velvetoptimiser</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+
+    <command><![CDATA[
+      export OMP_NUM_THREADS=2 &&
```
Comment (yhoogstrate) [inline tools/velvetoptimiser/velvetoptimiser.xml]: Does `=2` need to be replaced with `=\${GALAXY_SLOTS:-2}` like below?

File: tools/velvetoptimiser/velvetoptimiser.xml [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+          #if str($i.paired_type.paired_type_selector) == "single"
+
+            #if $i.__index__ > 1
+              -short$i.__index__
+            #else
+              -short
+            #end if
+
+            $i.paired_type.input1
```
Comment (yhoogstrate) [inline tools/velvetoptimiser/velvetoptimiser.xml]: Quotes?

File: tools/velvetoptimiser/velvetoptimiser.xml [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+          reference (for pre-mapped sam or bam files - see Velvet manual for details on how to use this option)
+
+
+    ]]></help>
+    <citations>
+        <citation type="bibtex">@UNPUBLISHED{GLADMAN2012,
+          author = "Gladman Simon",
+          title = "VelvetOptimiser",
+          year = "2012",
+          note = "https://github.com/slugger70/VelvetOptimiser"}
```
Comment (yhoogstrate) [inline tools/velvetoptimiser/velvetoptimiser.xml]: +`        url = "https://github.com/slugger70/VelvetOptimiser",`

**Pull request closed without merging**